src/Discord/Internal/Types/Events.hs

```diff
@@ -213,14 +214,25 @@ reparse val = case parseEither parseJSON $ toJSON val of
                 Left r -> fail r
                 Right b -> pure b
 
+-- | Remove the "wss://" and the trailing slash in a gateway URL, thereby returning
+-- the hostname portion of the URL that we can connect to.
+extractHostname :: String -> HostName
+extractHostname ('w':'s':'s':':':'/':'/':rest) = extractHostname rest
+extractHostname ('/':[]) = []
+extractHostname (a:b) = a:(extractHostname b)
+extractHostname [] = []
```

Comment:
Cheeky

I can't think of a simpler way without regex, which might not even be simpler.

Reply:
Haha, I considered what the best way to go about this was - especially since Discord doesn't clarify anywhere whether the URL can be assumed to always have the prefix/suffixes either. So I couldn't just do a drop/take too since they might not even be there.